crates/subspace-block-relay/src/consensus.rs

```diff
@@ -0,0 +1,582 @@
+//! Relay implementation for consensus blocks.
+
+use crate::protocol::compact_block::{CompactBlockClient, CompactBlockServer};
+use crate::protocol::{ProtocolBackend, ProtocolClient, ProtocolServer};
+use crate::utils::{RequestResponseStub, ServerMessage};
+use crate::{DownloadResult, RelayClient, RelayError, RelayServer, LOG_TARGET};
+use async_trait::async_trait;
+use codec::{Decode, Encode};
+use futures::channel::{mpsc, oneshot};
+use futures::stream::StreamExt;
+use lru::LruCache;
+use parking_lot::Mutex;
+use prost::Message;
+use sc_client_api::{BlockBackend, HeaderBackend};
+use sc_network::request_responses::{IncomingRequest, OutgoingResponse, ProtocolConfig};
+use sc_network::types::ProtocolName;
+use sc_network::{PeerId, RequestFailure};
+use sc_network_common::sync::message::BlockAttributes;
+use sc_network_sync::block_relay_protocol::{BlockDownloader, BlockRelayParams, BlockServer};
+use sc_network_sync::service::network::NetworkServiceHandle;
+use sc_network_sync::{BlockDataSchema, BlockRequestSchema, BlockResponseSchema, FromBlockSchema};
```

Comment:
@nazar-pc this is where/how the protobufs are used?

Reply:
You have a different protocol, there is no requirement to reuse the same data structures.

I think it would be easier for maintenance to switch to SCALE codec here and use enums for different data structures (so we can upgrade them in the future if necessary) rather than Substrate data structures generated from protobuf schemas. We can also remove unnecessary fields we don't have like justifications.

Reply:
OK, let me try changing the interface and update you. Closing this and https://github.com/subspace/substrate/pull/48 meanwhile